pkgs/unified_analytics/lib/src/event.dart

```diff
@@ -206,6 +206,36 @@ final class Event {
           if (exitCode != null) 'exitCode': exitCode,
         };
 
+  /// Event that contains the results for a specific doctor validator.
+  ///
+  /// [validatorName] - the name for the doctor validator.
+  ///
+  /// [result] - the final result for a specific doctor validator.
+  ///
+  /// [partOfGroupedValidator] - `true` indicates that this validator belongs
+  ///   to a grouped validator.
+  ///
+  /// [doctorInvocationId] - epoch formatted timestamp that can be used in
+  ///   combination with the client ID in GA4 to group the validators that
+  ///   ran in one doctor invocation.
+  ///
+  /// [statusInfo] - optional description of the result from the
+  ///   doctor validator.
+  Event.doctorValidatorResult({
+    required String validatorName,
+    required String result,
+    required bool partOfGroupedValidator,
+    required int doctorInvocationId,
+    String? statusInfo,
+  })  : eventName = DashEvent.doctorValidatorResult,
+        eventData = {
+          'validatorName': validatorName,
+          'result': result,
+          'partOfGroupedValidator': partOfGroupedValidator,
+          'doctorInvocationId': doctorInvocationId,
+          if (statusInfo != null) 'statusInfo': statusInfo,
+        };
```

Comment:
fyi @zanderso – continuing from our discussion, we can include fields such as `statusInfo` for future events that are marked as optional.. this can help reduce the time it takes for us to add additional information for an event without having to roll the latest version of this package

Reply:
Wondering if `statusInfo` can be a `Map<String, String>` that then gets encoded as json?

Reply:
Yeah we can probably have that kind of pattern for future events but for the doctor validators we only had one additional parameter I could add so I figured it would be easier to just send a string and not have to worry about parsing json on the backend

https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/doctor_validator.dart#L161

Unless we plan on adding more parameters to `ValidationResult` in the future?

Reply:
Ah, I see. `statusInfo` is a field that's already populated in the doctor validator results. `String?` for that makes sense. I was thinking of something more along the lines of the `Map<String, String>` that the current analytics API accepts that could be tacked on to any event to allow adding data as needed without a new unified_analytics package version.

Reply:
I would agree with what Zach's saying, except for the 100 character limit (that sample statusInfo is 81 chars :( )